pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -249,6 +255,74 @@ func (rc *reconciler) mountAttachedVolumes(volumeToMount cache.VolumeToMount, po
 	}
 }
 
+// processReconstructedVolumes checks volumes which were skipped during the reconstruction
+// process because it was assumed that since these volumes were present in DSOW they would get
+// mounted correctly and make it into ASOW.
+// But if mount operation fails for some reason then we still need to mark the volume as uncertain
+// and wait for the next reconciliation loop to deal with it.
+func (rc *reconciler) processReconstructedVolumes() {
+	if rc.kubeClient != nil {
+		rc.updateDevicePath(rc.skippedDuringReconstruction)
+	}
+	for volumeName, glblVolumeInfo := range rc.skippedDuringReconstruction {
+		// check if volume is marked as attached to the node
+		// for now lets only process volumes which are at least known as attached to the node
+		// this should help with most volume types (including secret, configmap etc)
+		if !rc.actualStateOfWorld.VolumeExists(volumeName) {
```

Comment:
so we might still miss some cases that volumes that are attachable plugin, if here it is not marked as attached yet, and later mount failed?

Reply:
I think if volume was attachable and node still holds the status in `volumeAttached` then - it should not be a problem. Could there be a case where volume is not even attached to the node and still requires cleanup? I think if someone force detached the volume then yeah we won't be cleaning the volume. I think it seems like a very edge case.

Reply:
if kubelet restarts, when volume will be marked as exist (attached) in actual state?

Reply:
I have changed the code in `processReconstructedVolumes` to wait for volume to be marked as attached via `VerifyControllerAttached` function. This means that `processReconstructedVolumes` will be called as long as one or more volumes remain in `skippedDuringReconstruction` map.

Reply:
could you point me where the wait logic?

Reply:
The wait logic is - the volume is only removed from `skippedDuringReconstruction` if it exists in ASOW.  So for volumes which were added to `skippedDuringReconstruction` during reconstruction and if volume does not exist (not even verified as attached) then volume is not removed from `skippedDuringReconstruction` and `processReconstructedVolumes` is called again for them.

Reply:
but when volume will be verified as attached? Only volume is in desired state already? If volume cannot be placed in desired state, it will not added as attached?

Reply:
volume can be in desired state temporarily, but it might be removed quickly, and volume is not added in desired state.